code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -28,28 +39,49 @@
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
+		var/newpassword = sanitize(input(usr, "Enter a password or leave blank to leave unprotected:"))
 		var/datum/computer_file/data/F = new/datum/computer_file/data()
 		F.filename = newname
 		F.filetype = "TXT"
+		if (newpassword)
+			F.password = newpassword
 		HDD.store_file(F)
 	if(href_list["PRG_deletefile"])
 		. = 1
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
 		var/datum/computer_file/file = HDD.find_file_by_name(href_list["PRG_deletefile"])
+		var/checkpass = ""
+		var/currentpass = file.password
 		if(!file || file.undeletable)
 			return 1
-		HDD.remove_file(file)
+		if(!currentpass)
+			HDD.remove_file(file)
+		else
+			checkpass = sanitize(input(usr, "Please enter the password:"))
+			if(checkpass == currentpass)
+				HDD.remove_file(file)
+			else
+				return 1
 	if(href_list["PRG_usbdeletefile"])
 		. = 1
-		var/obj/item/weapon/computer_hardware/hard_drive/RHDD = computer.portable_drive
+		var/obj/item/weapon/computer_hardware/hard_drive/RHDD = computer.hard_drive
```

Comment:
This change seems weird. Can you explain the difference between portable_drive and hard_drive? It could lead to odd things.

Reply:
This was unintended,